jaeger-core/src/main/java/com/uber/jaeger/senders/HttpSender.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2016, Uber Technologies, Inc
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.jaeger.senders;
+
+import com.uber.jaeger.thriftjava.Batch;
+import com.uber.jaeger.thriftjava.Collector.submitBatches_args;
+import com.uber.jaeger.thriftjava.Process;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.List;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ByteArrayEntity;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.thrift.TException;
+import org.apache.thrift.TSerializer;
+import org.apache.thrift.protocol.TBinaryProtocol.Factory;
+import org.apache.thrift.protocol.TProtocolFactory;
+
+/**
+ * @author Pavol Loffay
+ */
+public class HttpSender extends ThriftSender {
+  public static final int HTTP_COLLECTOR_PORT = 14268;
+  private static final TProtocolFactory PROTOCOL_FACTORY = new Factory();
+  private static final TSerializer SERIALIZER = new TSerializer(PROTOCOL_FACTORY);
+
+  private HttpClient httpClient = new DefaultHttpClient();
+  private URI collectorUri;
+
+  public HttpSender(String host, int maxPacketSize) {
+    super(PROTOCOL_FACTORY, maxPacketSize);
+    this.collectorUri = constructCollectorUri(host);
+    this.httpClient = new DefaultHttpClient();
+  }
+
+  @Override
+  public void send(Process process, List<com.uber.jaeger.thriftjava.Span> spans) throws TException {
+    submitBatches_args submitBatches =
```

Comment:
It refers to https://github.com/uber/jaeger/blob/master/cmd/collector/app/handler.go#L77 cc @badiib